processing/src/main/java/io/druid/granularity/PeriodGranularity.java

```diff
@@ -330,6 +321,43 @@ private long truncateMillisPeriod(final long t)
   }
 
   @Override
+  public Integer compare(QueryGranularity other)
+  {
+    if (other instanceof PeriodGranularity) {
+      PeriodGranularity pg1 = this;
+      PeriodGranularity pg2 = (PeriodGranularity) other;
+      Chronology c1 = pg1.getChronology();
+      Chronology c2 = pg2.getChronology();
+      if (pg1.getOrigin() != pg2.getOrigin() || !c1.equals(c2)) {
+        // todo: compare granularities with different origins
+        return null;
+      }
+      Period p1 = pg1.getPeriod();
+      Period p2 = pg2.getPeriod();
+      if (!isPrecise(p1, c1) || !isPrecise(p2, c2)) {
+        return null;
+      }
+      return compareMultiplicity(p1.toStandardDuration().getMillis(), p2.toStandardDuration().getMillis());
+    }
+    return super.compare(other);
+  }
+
+  private boolean isPrecise(Period period, Chronology chronology)
```

Comment:
I've copied javadoc of `DruationField#isPrecise()`. Time related things are really confusing.